frost-core/src/frost/keys.rs

```diff
@@ -156,11 +178,58 @@ pub(super) struct CoefficientCommitment<C: Ciphersuite>(pub(super) Element<C>);
 /// [`VerifiableSecretSharingCommitment`], either by performing pairwise comparison, or by using
 /// some agreed-upon public location for publication, where each participant can
 /// ensure that they received the correct (and same) value.
-#[derive(Clone)]
+#[derive(Clone, PartialEq)]
 pub struct VerifiableSecretSharingCommitment<C: Ciphersuite>(
-    pub(super) Vec<CoefficientCommitment<C>>,
+    pub(crate) Vec<CoefficientCommitment<C>>,
 );
 
+impl<C> VerifiableSecretSharingCommitment<C>
+where
+    C: Ciphersuite,
+{
+    /// Returns concatenated coefficent commitments from self.0 specifying the number of commitments
+    pub fn serialize(&self) -> String {
+        let values = self
+            .0
+            .iter()
+            .map(|cc| hex::encode(<<C as Ciphersuite>::Group as Group>::serialize(&cc.0)))
+            .join("");
+
+        hex::encode(self.0.len().to_string()) + &values
+    }
+
+    /// Returns VerifiableSecretSharingCommitment from a vector of CoefficientCommitments
+    pub fn deserialize(coefficientCommitments: String) -> Result<Self, Error<C>> {
+        let decoded = hex::decode(&coefficientCommitments)
+            .map_err(|_| Error::from(GroupError::MalformedElement))?;
+
+        let coeff_commitments_data = &decoded[1..decoded.len()].to_vec();
+
+        let n: usize = String::from_utf8([decoded[0]].to_vec())
+            .map_err(|_| Error::InvalidCoefficient)?
+            .parse()
+            .map_err(|_| Error::InvalidCoefficient)?;
+        let l = coeff_commitments_data.len() / n;
+
+        let mut coeff_commitments: Vec<CoefficientCommitment<C>> = Vec::with_capacity(n);
+
+        for i in 0..n {
+            let commitment_value = hex::encode(&coeff_commitments_data[(i * l)..((i * l) + l)]);
+            println!("comm value: {}", commitment_value);
+            let serialized: <C::Group as Group>::Serialization =
+                <C::Group as Group>::Serialization::try_from(
+                    hex::decode(commitment_value)
+                        .map_err(|_| Error::from(GroupError::MalformedElement))?,
+                )
+                .map_err(|_| Error::InvalidCoefficient)?;
+            let coeff_commitment = CoefficientCommitment::<C>::deserialize(serialized)?;
+
+            coeff_commitments.push(coeff_commitment)
+        }
+        Ok(Self(coeff_commitments))
+    }
+}
```

Comment:
Similarly this will require changing the function to `pub fn deserialize(coefficientCommitments: &[<C::Group as Group>::Serialization])` and changing the body accordingly (it will become much simpler). Likewise you can use the encoding logic in the demo instead.